### PR TITLE
Add ballot ready to data warehouse

### PR DESCRIPTION
**Pull request by bendnorman**
This PR adds the ballot-ready data to the data warehouse and some information for the next election in each county.

### CODE REVIEW

**Comment by bendnorman** [comment]
Running into some weird errors:
1. The ballot ready data has an election in Yakatat that is not present in the main `counties_wide_format` data frame.
2. I'm getting this validation error because there are about a hundred new counties mostly from Puerto Rico. Not sure how the changes in the PR would have introduced counties:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/app/dbcp/cli.py", line 95, in <module>
    sys.exit(main())
  File "/app/dbcp/cli.py", line 91, in main
    dbcp.data_mart.create_data_marts(args)
  File "/app/dbcp/data_mart/__init__.py", line 70, in create_data_marts
    validate_data_mart(engine=engine)
  File "/app/dbcp/validation/tests.py", line 245, in validate_data_mart
    test_county_long_vs_wide(engine)
  File "/app/dbcp/validation/tests.py", line 205, in test_county_long_vs_wide
    n_counties_wide == n_counties_long
AssertionError: counties_wide_format and counties_long_format have different county coverage
make: *** [all_local] Error 1
```

**Comment by TrentonBush** [comment]
That test checks that county_wide and county_long have consistent spatial coverage.
1. Yakatat's FIPS code (02261) last exists in the 2010 FIPS vintage. The data mart currently uses 2020 vintage and drops anything from earlier/later (very few items, but that is one)
2. The ballot ready stuff was only integrated into `county_wide`. I moved it to the `_get_county_properties()` function, which is used by both `county_wide` and `county_long` constructors (and added the new columns to the metadata). The test also relies on using that function to identify and drop county-level columns so that only spatial coverage of technical data is compared. Now the test passes.

**Comment by bendnorman** [comment]
Ok the gitbook has been updated, the data is in `data_mart_dev` and is updated with the August version of the raw data.

**Review by TrentonBush — CHANGES_REQUESTED** [review]
This data looks like it could be normalized into at least 4 tables: race, election, position, and a race-county m:m association table  (maybe more association tables - I didn't look at the other relationships). Given the limited scope of the client's uses for this data, I think it's probably fine that we leave it denormalized and save some effort. But if we do that, let's make a `data_mart` table for end users, even if it is just `select *` from this table. That way we:
1. can make "business logic" or UI changes in SQL instead of the transform function (such as their request to replace state abbreviations with full names)
1. preserve the option to normalize the warehouse level table later if they find other uses for this data (no API consistency commitments)

Can you make a `data_mart` table that joins on the standardized `data_warehouse.county_fips.county_name` and `data_warehouse.state_fips.state_name` fields?

**Comment by bendnorman** [comment]
Yeah this could for sure be normalized into multiple tables. I didn't normalize to speed up integration. I'll create a data mart table with the normalized data. Once the data is in BQ, I'll go back an normalize the data warehouse table.

**Comment by bendnorman** [comment]
I made the requested changes:
- new data mart table called `br_election_data` that is a copy of the data warehouse table but without the `raw_` prefixes.
- new data mart table called `county_commission_election_info` that has county commission elections for each county.

**Review by TrentonBush — COMMENTED** [review]
Looks good! I just had a couple questions about the data itself and one straggling piece of county code.

File: src/dbcp/data_mart/counties.py [hunk 1]
```diff
@@ -916,6 +916,32 @@ def _get_actionable_aggs_for_wide_format(engine: sa.engine.Engine) -> pd.DataFra
     return aggs
 
 
+def _get_ballot_ready_election_cols(engine: sa.engine.Engine) -> pd.DataFrame:
```
[the remark this answers is not in the document]
Comment (bendnorman) [reply src/dbcp/data_mart/counties.py:919]: Ah thank you. Will remove.

**Review by TrentonBush — APPROVED** [review]

**Pull request merged**